core/src/main/java/org/jdbi/v3/core/argument/PrimitivesArgumentFactory.java

```diff
@@ -27,4 +44,43 @@
         register(float.class, Types.FLOAT, PreparedStatement::setFloat);
         register(double.class, Types.DOUBLE, PreparedStatement::setDouble);
     }
+
+    @Override
+    public Optional<Argument> build(Type expectedType, Object value, ConfigRegistry config) {
+        Class<?> expectedClass = getErasedType(expectedType);
+
+        if (value != null && expectedClass == Object.class) {
+            expectedClass = value.getClass();
+        }
+
+        return Optional.ofNullable(builders.get(expectedClass)).map(r -> r.apply(value, config));
+    }
+
+    @Override
+    public Optional<Function<Object, Argument>> prepare(Type type, ConfigRegistry config) {
+        return Optional.empty();
+    }
+
+    @Override
+    public Collection<? extends Type> prePreparedTypes() {
+        return Collections.emptySet();
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    <T> void register(Class<T> klass, int sqlType, StatementBinder<T> binder) {
+        builders.put(klass, (value, config) -> {
+            if (value != null) {
```

Comment:
Most of this lambda's functionality is duplicated from the super class.  Can we wrap the `binder` argument with the new null check and call `super.register` to handle the existing functionality?  That would be much more in line with the expected use of the abstract superclass.